osquery/tables/system/windows/windows_search.cpp

```diff
@@ -0,0 +1,244 @@
+    case VT_I2:
+      wss << variant.iVal;
+      break;
+    case VT_UI4:
+      wss << variant.ulVal;
+      break;
+    case VT_I4:
+      wss << variant.lVal;
+      break;
+    case VT_UI8:
+      wss << variant.uhVal.HighPart << variant.uhVal.LowPart;
+      break;
+    case VT_I8:
+      wss << variant.hVal.HighPart << variant.hVal.LowPart;
+      break;
+    case VT_DATE:
+      wss << dateToUnixTime(variant.date);
+      break;
+    default:
+      wss << "unhandled variant type " << variant.vt;
+      break;
+    }
+  }
+}
+
+std::string ccomandColumnStringValue(
+    CCommand<CDynamicAccessor, CRowset>& cCommand, DBORDINAL columnIndex) {
+  DBTYPE type;
+  cCommand.GetColumnType(columnIndex, &type);
+
+  std::wstringstream wss;
+
+  DBSTATUS status;
+  cCommand.GetStatus(columnIndex, &status);
+  if (status == DBSTATUS_S_ISNULL) {
+    wss << "NULL";
+  } else if (status == DBSTATUS_S_OK || status == DBSTATUS_S_TRUNCATED) {
+    DBTYPE type;
+    cCommand.GetColumnType(columnIndex, &type);
+    switch (type) {
+    case DBTYPE_VARIANT:
+      writePropVariant(
+          *(static_cast<PROPVARIANT*>(cCommand.GetValue(columnIndex))), wss);
+      break;
+    case DBTYPE_WSTR: {
+      DBLENGTH cbLen;
+      cCommand.GetLength(columnIndex, &cbLen);
+      WCHAR szBuffer[2048];
+      StringCchCopyN(szBuffer,
+                     ARRAYSIZE(szBuffer),
+                     static_cast<WCHAR*>(cCommand.GetValue(columnIndex)),
+                     cbLen / sizeof(WCHAR));
+      wss << szBuffer;
+    } break;
+    case DBTYPE_I1:
+      wss << *static_cast<UCHAR*>(cCommand.GetValue(columnIndex));
+      break;
+    case DBTYPE_UI2:
+      wss << *static_cast<USHORT*>(cCommand.GetValue(columnIndex));
+      break;
+    case DBTYPE_I2:
+      wss << *static_cast<SHORT*>(cCommand.GetValue(columnIndex));
+      break;
+    case DBTYPE_UI4:
+      wss << *static_cast<DWORD*>(cCommand.GetValue(columnIndex));
+      break;
+    case DBTYPE_I4:
+      wss << *static_cast<INT*>(cCommand.GetValue(columnIndex));
+      break;
+    case DBTYPE_UI8:
+      wss << *static_cast<ULONGLONG*>(cCommand.GetValue(columnIndex));
+      break;
+    case DBTYPE_I8:
+      wss << *static_cast<LONGLONG*>(cCommand.GetValue(columnIndex));
+      break;
+    case DBTYPE_DATE:
+      wss << dateToUnixTime(
+          *static_cast<DATE*>(cCommand.GetValue(columnIndex)));
+      break;
+    default:
+      wss << "unhandled database type " << type;
+      break;
+    }
+  } else {
+    wss << "error reading column";
+  }
+
+  std::wstring wideStr = wss.str();
+  std::wstring_convert<std::codecvt_utf8<wchar_t>> converter;
+  return converter.to_bytes(wideStr);
+}
+
+std::vector<std::map<std::string, std::string>> executeWindowsSearchQuery(
+    CSession& cSession, const std::string& query) {
+  HRESULT hr;
+  std::vector<std::map<std::string, std::string>> results;
+
+  CCommand<CDynamicAccessor, CRowset> cCommand;
+  hr = cCommand.Open(cSession, query.c_str());
```

Comment:
The query parameter should support AQS. I'm going to expand on a PR comment, but it would be good if this table have a fixed number of item properties to display. You can do this by using the query helper - see [here](https://github.com/microsoft/Windows-classic-samples/blob/7cbd99ac1d2b4a0beffbaba29ea63d024ceff700/Samples/Win7Samples/winui/WindowsSearch/WSOleDB/WSOleDB.cpp#L51) and [here](https://learn.microsoft.com/en-us/windows/win32/api/searchapi/nn-searchapi-isearchqueryhelper) for more information.

Reply:
added AQS support!